buildbot_nix/__init__.py

```diff
@@ -816,14 +817,13 @@ def run(self) -> Generator[Any, object, Any]:
         if not out_path:  # if, e.g., the build fails and doesn't produce an output
             return util.SKIPPED
 
-        project_name = Path(props.getProperty("projectname"))
-
-        target = Path(props.getProperty("branch"))
-
-        attr = Path(props.getProperty("attr"))
+        owner = urllib.parse.quote_plus(self.project.owner)
+        repo = urllib.parse.quote_plus(self.project.name)
+        target = urllib.parse.quote_plus(props.getProperty("branch"))
+        attr = urllib.parse.quote_plus(props.getProperty("attr"))
```

Comment:
This on its own, should already be filesystem safe... The test below is just a safety check.